src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -475,67 +477,57 @@ private void Poll()
+                foreach (var tpo in _seekedOffset.Select(kvp => kvp.Value))
                 {
-                    // Seek has to be done here because they can somehow fail.
-                    // Would need to see if we can move this somewhere else
-                    // because a seek can take up to 200ms to complete
-                    foreach (var tpo in _seekedOffset.Select(kvp => kvp.Value))
+                    try
                     {
-                        try
-                        {
-                            if(_log.IsDebugEnabled)
-                                _log.Debug("Seeking offset {0} in partition {1} for topic {2}", tpo.Offset, tpo.Partition, tpo.Topic);
-                            _consumer.Seek(tpo);
-                        }
-                        catch (Exception ex)
-                        {
-                            _log.Error(ex, $"{tpo.TopicPartition} Failed to seek to {tpo.Offset}: {ex}");
-                            throw;
-                        }
+                        if(_log.IsDebugEnabled)
+                            _log.Debug("Seeking offset {0} in partition {1} for topic {2}", tpo.Offset, tpo.Partition, tpo.Topic);
+                        _consumer.Seek(tpo);
                     }
-                    
-                    // resume partitions to fetch
-                    IImmutableSet<TopicPartition> partitionsToFetch = _requests.Values.SelectMany(v => v.Topics).ToImmutableHashSet();
-                    var resumeThese = currentAssignment.Where(partitionsToFetch.Contains).ToList();
-                    var pauseThese = currentAssignment.Except(resumeThese).ToList();
-                    PausePartitions(pauseThese);
-                    ResumePartitions(resumeThese);
-
-                    using (var cts = new CancellationTokenSource(_settings.PollTimeout))
+                    catch (Exception ex)
                     {
-                        ProcessResult(partitionsToFetch, PollKafka(cts.Token));
+                        _log.Error(ex, $"{tpo.TopicPartition} Failed to seek to {tpo.Offset}: {ex}");
+                        throw;
                     }
                 }
+                
+                // resume partitions to fetch
+                IImmutableSet<TopicPartition> partitionsToFetch = _requests.Values.SelectMany(v => v.Topics).ToImmutableHashSet();
+                var resumeThese = currentAssignment.Where(partitionsToFetch.Contains).ToList();
+                var pauseThese = currentAssignment.Except(resumeThese).ToList();
+                PausePartitions(pauseThese);
+                ResumePartitions(resumeThese);
+
+                using (var cts = new CancellationTokenSource(_settings.PollTimeout))
+                {
+                    var (polled, exception) = PollKafka(cts.Token);
+                    ProcessResult(partitionsToFetch, polled);
+                    ProcessExceptions(exception);
```

Comment:
Process all messages that happened before we hit any exceptions, then process the exception, if any.